### PR TITLE
Merge in SubSamplingProcess and supporting changes

### PR DESCRIPTION
**Pull request by turbomam**
- #1198 

will add commits soon

### CODE REVIEW

**Comment by turbomam** [comment]
see 
- https://github.com/microbiomedata/nmdc-schema/pull/1201

**Comment by JamesTessmer** [comment]
Added:
SubSamplingProcess class to [nmdc.yaml](https://github.com/microbiomedata/nmdc-schema/blob/1198-merge-subsampling-and-extraction-classes-with-data-that-illustrates-the-relationship-into-nmdc-schema/src/schema/nmdc.yaml)
Valid [data file](https://github.com/microbiomedata/nmdc-schema/blob/1198-merge-subsampling-and-extraction-classes-with-data-that-illustrates-the-relationship-into-nmdc-schema/src/data/valid/SubSamplingProcess-minimal.yaml)

Modified:
[sample_prep.yaml](https://github.com/microbiomedata/nmdc-schema/blob/1198-merge-subsampling-and-extraction-classes-with-data-that-illustrates-the-relationship-into-nmdc-schema/src/schema/sample_prep.yaml) due to slot definition conflicts, and previous discussion that we would redefine slots from sample_prep.

File: src/schema/nmdc.yaml [hunk 1]
```diff
@@ -2781,3 +2808,22 @@ slots:
...
+  container_size:
+    range: QuantityValue
+    description: The volume of the container an analyte is stored in or an activity takes place in
+
+  container_type:
+    range: ContainerTypeEnum
+    description: >-
+      A type of container. i.e. test tube, falcon tube, etc.
+
+  volume:
```
Comment (cmungall) [inline src/schema/nmdc.yaml:2821]: give this a name that resolves ambiguity as to what volume is being referred to

File: src/schema/nmdc.yaml [hunk 2]
```diff
@@ -2781,3 +2808,22 @@ slots:
...
+    range: QuantityValue
+    description: The volume of the solvent/solute being used, not the input.
```
Comment (cmungall) [inline src/schema/nmdc.yaml:2823]: Here we are defining a very general schema-wide slot `volume`; we can imagine this being used many places, e.g. volume of a container, of a piece of soil. But the description is very specific. Either do one of:

1. Make a more specific slot such as solvent_volume, and keep the definition
2. Keeping a generic volume, and amending this definition to be more generic, and putting the specific description on slot_usage in SubSamplingProcesss

**Review by cmungall — CHANGES_REQUESTED** [review]

**Comment by turbomam** [comment]
Thanks James. Ina addition to @cmungall's request, could you please also add micro-crediting for yourself or others into the elements you want to merge (classes, slots, enumeration).

**Comment by JamesTessmer** [comment]
Added the change to make volume more general, and add a more specific description in the slot usage. Also added credits for contributors.

**Pull request merged**